### ISSUE TITLE
gdf/arrow: missing support for decimal, date, time, timestamp

### ISSUE THREAD

**Issue by andrewseidl** [heavyai/heavydb#206]
We currently only support bool, int*, float*, and text when returning results using Arrow. Selecting decimal, date, time, or timestamp columns isn't currently supported and returns a somewhat confusing exception:

    <type> is not supported in temporary table.

Type convert and the data conversion/column builder is around https://github.com/mapd/mapd-core/blob/dd6c3c6482630f3794c6e4cbd9783e70fd4436f3/QueryEngine/ResultSetConversion.cpp#L151-L185

cc: https://github.com/mapd/pymapd/issues/75
cc: https://github.com/mapd/pymapd/issues/76

**Comment by xmnlab** [comment]
@andrewseidl 

I am reviewing mapdc-core and arrow-cpp source ... 

I have some guess about some  data types and I will appreciate any comment about that.

below there is a list of mapd type constant and an data type option from arrow [1]:
 
kDECIMAL: decimal_type(int precision, int scale)
kNUMERIC: could be the same of kDECIMAL?
kCHAR: maybe utf8()?
kVARCHAR: maybe utf8()?
kTIME: time32()
kTIMESTAMP: date64()
kDATE: date32()

does it make sense?

[1] https://github.com/apache/arrow/blob/master/cpp/src/arrow/type_fwd.h

**Comment by andrewseidl** [comment]
Yup, only one that you might want to look at closer is `kTIMESTAMP` since Arrow does have a Timestamp type http://arrow.apache.org/docs/cpp/classarrow_1_1_timestamp_type.html

`kDECIMAL` / `kNUMERIC`: Yes, they're actually the same internally, so can treat them both as `decimal(mapd_type.get_precision(), mapd_type.get_scale())`.

`kCHAR` / `kVARCHAR` / `kTEXT`: Yes, `utf8()` / strings. Same for kTEXT, but in all cases need to pay attention for dictionary encoding, which is already handled (but could be improved - it's currently pretty slow).

`kTIME` / `kDATE`: Yes, `time32(TimeUnit::SECOND)` / `date32()` (for now at least)

`kTIMESTAMP`: should probably instead be `timestamp(TimeUnit::SECOND)`

**Comment by randyzwitch** [comment]
Internal tickets: 
https://jira.omnisci.com/browse/BE-3403 
https://jira.omnisci.com/browse/BE-2037

**Comment by psilvest** [comment]
This issue has been inactive for a while, and the project has changed substantially. If this still occurs on the latest release, please add reproduction steps and environment details.
